### PR TITLE
Upgrading to Brighterscript v1

### PR DESCRIPTION
**Pull request by markwpearce**
Goals:

- [x] Upgrade all Brighterscript dependencies to the latest alpha version
- [x] Modify plugin to work with new plugin architecture
- [x] Make all Brighterscript code type safe and fix errors

Also:
- [x] clean up package.json and build foolishness
- [x] add ability to run Rooibos tests from command line

### CODE REVIEW

File: bsc-plugin/src/lib/rooibos/Utils.ts [hunk 1]
```diff
@@ -115,3 +129,12 @@ export function getScopeForSuite(testSuite: TestSuite) {
         return testSuite.file.program.getFirstScopeForFile(testSuite.file);
     }
 }
+
+export function getFileLookups(file: BrsFile): CachedLookups {
```
Comment (markwpearce) [inline bsc-plugin/src/lib/rooibos/Utils.ts:133]: New function to replace file.parser.references

File: bsc-plugin/src/lib/rooibos/RooibosSession.ts [hunk 2]
```diff
@@ -210,11 +210,11 @@ export class RooibosSession {
     createNodeFile(program: Program, suite: TestSuite) {
         let xmlText = this.getNodeTestXmlText(suite);
         this.fileFactory.addFile(program, suite.xmlPkgPath, xmlText);
-        let bsFile = program.getFile(suite.bsPkgPath);
-        if (bsFile) {
-            (bsFile as BrsFile).parser.statements.push();
-            bsFile.needsTranspiled = true;
-        }
+        //let bsFile = program.getFile(suite.bsPkgPath);
```
Comment (markwpearce) [inline bsc-plugin/src/lib/rooibos/RooibosSession.ts]: I don't think this is needed anymore in v1

File: bsc-plugin/src/lib/utils/Diagnostics.ts [hunk 3]
```diff
@@ -14,7 +14,10 @@ function addDiagnostic(
     severity: DiagnosticSeverity = DiagnosticSeverity.Error
 ) {
     endLine = endLine === -1 ? startLine : endLine;
-    file.addDiagnostics([createDiagnostic(file, code, message, startLine, startCol, endLine, endCol, severity)]);
+    file.program.diagnostics.register(
```
Comment (markwpearce) [inline bsc-plugin/src/lib/utils/Diagnostics.ts]: Using new diagnostics manager

File: bsconfig.json [hunk 4]
```diff
@@ -18,6 +17,6 @@
     }
   ],
   "plugins": [
-    "@rokucommunity/bslint"
```
Comment (markwpearce) [inline bsconfig.json:21]: temporarily removed until bslint alpha-35 release

File: framework/src/source/BaseTestSuite.bs [hunk 5]
```diff
@@ -15,18 +15,18 @@ namespace rooibos
     protected hasIgnoredTests = false
     protected isSolo = false
     protected isIgnored = false
-    protected noCatch = false
+    public noCatch = false
```
Comment (markwpearce) [inline framework/src/source/BaseTestSuite.bs:19]: Many changes in the .BS files to support proper typing and to fix validation errors

File: tests/bsconfig.json [hunk 6]
```diff
@@ -3,7 +3,8 @@
   "files": [
     "manifest",
     "source/**/*.*",
-    "components/**/*.*"
+    "components/**/*.*",
+    { "src": "../../framework/src/source/**/*", "dest": "source/rooibos" }
```
Comment (markwpearce) [inline tests/bsconfig.json]: Be able to import the framework in the tests

File: tests/src/source/Assertion.spec.bs [hunk 7]
```diff
@@ -11,7 +12,7 @@ namespace tests
     '+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 
     @it("Fail")
-    function _()
+    function _() 'bs:disable-line 1026
```
Comment (markwpearce) [inline tests/src/source/Assertion.spec.bs]: All the functions are `function _()` which causes validation error 1026 - `Duplicate identifier`

File: bsconfig.json (hunk 4 above)
Comment (TwitchBronBron) [reply bsconfig.json:21]: I just released bslint alpha-35, so you should be good-to-go now.

File: tests/src/source/Assertion.spec.bs (hunk 7 above)
Comment (TwitchBronBron) [reply tests/src/source/Assertion.spec.bs]: I think the plugin itself needs to suppress this one. Writing all the functions as `_` is a "feature" of this test framework, so it should disable those diagnostics internally so devs don't need to add this comment everywhere (defeating the entire point of naming the functions `_` in the first place)

**Pull request merged**